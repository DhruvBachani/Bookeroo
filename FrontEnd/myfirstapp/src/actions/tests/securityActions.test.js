import React from 'react';
import reducer from '../../../src/reducers'
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();
const store = mockStore([]);

const login = () => ({
    type: 'SET_CURRENT_USER'
});

const logout = () => ({
    type: 'SET_CURRENT_USER'
});

describe('securityActions', () => {
    it('creates SET_CURRENT_USER when logging in', () => {
        store.dispatch(login());
        expect(store.getActions()).toMatchSnapshot();
    })

    it('creates SET_CURRENT_USER when logging out', () => {
        store.dispatch(logout());
        expect(store.getActions()).toMatchSnapshot();
    })
})