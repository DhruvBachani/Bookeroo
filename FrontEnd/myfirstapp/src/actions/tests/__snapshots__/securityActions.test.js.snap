// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`securityActions creates SET_CURRENT_USER when logging in 1`] = `
Array [
  Object {
    "type": "SET_CURRENT_USER",
  },
]
`;

exports[`securityActions creates SET_CURRENT_USER when logging out 1`] = `
Array [
  Object {
    "type": "SET_CURRENT_USER",
  },
  Object {
    "type": "SET_CURRENT_USER",
  },
]
`;
